blueoil/datasets/fer_2013.py

```diff
@@ -0,0 +1,142 @@
+# -*- coding: utf-8 -*-
+# Copyright 2020 The Blueoil Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# =============================================================================
+import functools
+import os
+import csv
+from glob import glob
+
+import numpy as np
+
+from blueoil.datasets.base import Base
+from blueoil.utils.random import shuffle
+
+
+def _load_csv(path):
+    with open(path) as csvfile:
+        reader = csv.reader(csvfile)
+        lines = [row for row in reader]
+    return lines[1:]  # pass the header
+
+
+class FER2013(Base):
+    classes = [
+        "Angry",
+        "Disgust",
+        "Fear",
+        "Happy",
+        "Sad",
+        "Surprise",
+        "Neutral"
+    ]
+    image_size = 48
+    num_classes = len(classes)
+    available_subsets = ["train", "test"]
+    extend_dir = "FER2013"
+
+    def __init__(
+        self,
+        subset="train",
+        batch_size=100,
+        *args,
+        **kwargs
+    ):
+        super().__init__(
+            subset=subset,
+            batch_size=batch_size,
+            *args,
+            **kwargs
+        )
+        self.images, self.labels = self._images_and_labels()
+
+    @property
+    def num_classes(self):
+        """Returns the number of classes"""
+        return len(self.classes)
+
+    @property
+    def num_per_epoch(self):
+        """Returns the number of datas in the data subset"""
+
+        return len(self.images)
+
+    @functools.lru_cache(maxsize=None)
+    def _load_data(self, path):
+        # Load the label and image data from csv
+        lines = _load_csv(path)
+        # private test data is used for user's test
+        train = [line for line in lines if line[2] == "Training"]
+        public_test = [line for line in lines if line[2] == "PublicTest"]
```

Comment:
This 2 line iterate `lines` twice.
Using `if else` statement to reduce iteration is better.